### DISCUSSION TITLE
Difference between TppObject and TppCertificate

### DISCUSSION THREAD

**Discussion by fearnie85** [gdbarron/VenafiTppPS#141] [Q&A]
What is the difference between these two?

I keep getting 500 server error when trying to generate a certificate list and not with tppobject.

Just wondering what is the difference?

Thanks,

**Comment by gdbarron** [comment]
TppObject is a generic class that all objects use.  TppCertificate, as a class, doesn't exist.  If you are referring to various functions with those names, TppObject acts on all objects whereas TppCertificate only acts on certificates.  Some of the functions have expanded capabilities depending on the object in question, eg. certificate.

**Reply by fearnie85** [reply]
thanks for your explaination.

what would be a good example of using tppobject? thanks,

**Reply by gdbarron** [reply]
Most of the functions use TppObject. For instance, `Find-TppObject` returns objects of type TppObject. You can see this by running that function and piping to get-member. Other functions accept TppObject as an input. I've taken lots of time to document the help so it should help point these out.
